coral-common/src/main/java/com/linkedin/coral/common/transformers/SqlCallTransformer.java

```diff
@@ -0,0 +1,98 @@
+/**
+ * Copyright 2017-2023 LinkedIn Corporation. All rights reserved.
+ * Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.coral.common.transformers;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.sql.SqlCall;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.SqlNodeList;
+import org.apache.calcite.sql.SqlSelect;
+import org.apache.calcite.sql.validate.SqlValidator;
+
+
+/**
+ * Abstract class for generic transformations on SqlCalls
+ */
+public abstract class SqlCallTransformer {
+  private SqlValidator sqlValidator;
+  private final List<SqlSelect> topSelectNodes = new ArrayList<>();
+
+  public SqlCallTransformer() {
```

Comment:
are we adding an empty constructor here to support Standard Sqlcall transformers where a sqlValidator is not required?

Reply:
Yes, it's needed for transformers without SqlValidator.